### ISSUE TITLE
Checkpoints store graph-attached loss tensors; `dec` eval runs without `no_grad`

### ISSUE THREAD

**Issue by p3zo** [p3zo/rhythmic-relationships#28]
Two small things in the same area.

**1. Checkpoints store a graph-attached loss tensor.**

```python
save_checkpoint(..., loss=loss, ...)   # loss is the live tensor from the batch loop
```

`torch.save` then serialises the tensor along with whatever of the autograd graph it references. Sites: `hits_encdec/train.py:120`, `encdec/train.py:120`, `chroma/train.py:120`, `hits_encdec_polydrums/train.py:568`, `hits_dec/train.py:284`, `hits_dec_xformer/train.py:252`, `rsp_fc/train.py:252`, `tvae/train.py:326`.

Pass `loss.item()`.

Related: `loss` is read after the batch loop ends, so an empty loader gives `NameError` rather than a clear message.

**2. Eval losses computed with gradients enabled.**

```python
# scripts/modeling/dec/train.py:107-119
for k in range(n_eval_iters):
    x, y = parse_sequential_batch(next(iter(train_loader)), device)
    logits = model(x)          # no torch.no_grad()
    loss = compute_loss(logits, y, loss_fn)
```

Builds a graph for every eval batch. The other scripts wrap this in `torch.no_grad()`; `dec` doesn't.
